File: mandacare-api/src/main/java/cm/mandacare/api/module/patient/ConsultationDecision.java
package cm.mandacare.api.module.patient;

public enum ConsultationDecision {
    KEEP_IN_CONSULTATION(VisitStatus.IN_CONSULTATION),
    SEND_TO_LAB(VisitStatus.CASH_DESK),
    RELEASE_PATIENT(VisitStatus.CASH_DESK);

    private final VisitStatus visitStatus;

    ConsultationDecision(VisitStatus visitStatus) {
        this.visitStatus = visitStatus;
    }

    VisitStatus visitStatus() {
        return visitStatus;
    }

    VisitStatus statusAfterCashDesk() {
        return switch (this) {
            case SEND_TO_LAB -> VisitStatus.LAB;
            case RELEASE_PATIENT -> VisitStatus.RELEASED;
            case KEEP_IN_CONSULTATION -> VisitStatus.IN_CONSULTATION;
        };
    }
}

File: mandacare-mobile/lib/features/consultations/domain/consultation_decision.dart
enum ConsultationDecision {
  keepInConsultation('Continuer', 'KEEP_IN_CONSULTATION'),
  sendToLab('Labo', 'SEND_TO_LAB'),
  releasePatient('Sortie', 'RELEASE_PATIENT');

  const ConsultationDecision(this.label, this.apiValue);

  final String label;
  final String apiValue;
}
